Bug 1197444: Rename kuma/static to kuma/assets

openjck opened the pull request:
It's easy to confuse the kuma/static directory and the root static
directory, especially when the term kuma is already in the path.

Can you easily tell the difference between these?

    /path/to/dev/folder/kuma/kuma/static
    /path/to/dev/folder/kuma/static

Bleh. This change renames kuma/static to kuma/assets to make the
distinction more clear.

--

To test:

* Browse around and ensure the site generally looks good
* Repeat after [enabling production assets](https://kuma.readthedocs.org/en/latest/development.html#production-assets)

mdn-services commented on kuma/assets/css/font-awesome.css:
```diff
@@ -0,0 +1,5 @@
+ *  Font Awesome 4.1.0 by @davegandy - http://fontawesome.io - @fontawesome
+ *  License - http://fontawesome.io/license (Font: SIL OFL 1.1, CSS: MIT License)
+ */
+@font-face{font-family:'FontAwesome';src:url("/static/css/libs/font-awesome/fonts/fontawesome-webfont.eot?v=4.1.0");src:url("/static/css/libs/font-awesome/fonts/fontawesome-webfont.eot?#iefix&v=4.1.0") format('embedded-opentype'),url("/static/css/libs/font-awesome/fonts/fontawesome-webfont.woff?v=4.1.0") format('woff'),url("/static/css/libs/font-awesome/fonts/fontawesome-webfont.ttf?v=4.1.0") for ...
```
Font unicode-range subsetting not supported by: IE (10,11), Firefox (38,39), Safari (8,7.1), iOS Safari (8,8.1-8.4), IE Mobile (10,11)

mdn-services commented on kuma/assets/css/wiki-wysiwyg.css:
```diff
@@ -0,0 +1,12 @@
+pre[class^='brush:python']:before, pre[class^='brush: python']:before { content: 'Python' }
+pre[class^='brush:sql']:before, pre[class^='brush: sql']:before { content: 'SQL' }
+pre[class^='brush:xml']:before, pre[class^='brush: xml']:before { content: 'XML' }
+li[data-default-state='open'] > a:after{content:'(defaults to "open")';color:#666;font-size:12px;font-size:.857rem;font-family:arial,sans-serif;font-style:italic;display:inline-block;padding-left:6px}details{position:relative;display:block;border:2px dotted #000;padding:3px;}details:before{font-size:10px;font-size:.714rem;content:'details';position:absolute;top:-17px;left:-2px;z-index:10;display: ...
```
rem (root em) units not supported by: IE (10)

mdn-services replied:
rem (root em) units not supported by: IE (10)

openjck commented:
Heh. Discord added some comments.

This change just moves the files, so I'm not worried about them.

groovecoder commented:
Checking this out now. And yes, the static thing is totally confusing to me too.

groovecoder commented:
On a simple checkout + runserver, I get 404s for all the CSS at first. Had to run `./scripts/compile-stylesheets` first. That will teach me to use foreman I guess.

groovecoder commented:
Unrelated to this PR(?) - mozorg CDN is missing a bunch of .woff2 files: http://screencast.com/t/kGW9xHwCL

groovecoder commented:
Intern tests run fine in both local static mode and production static mode. (Nice way to verify was to see the requests for the static assets show up in runserver when running with `DEBUG=True` and then see them disappear when running with `DEBUG=False`).

nits?

* Our [production assets docs instruction to restart apache](https://kuma.readthedocs.org/en/latest/development.html#production-assets) could be explicit: `sudo apache2ctl restart` ?

* I noticed I have/had a `static/assets/` directory in my box, but it seems unnecessary? I deleted it completely and all tests still run fine in both modes. Is there some `static/` cleanup to do after this change?

openjck commented:
> Unrelated to this PR(?) - mozorg CDN is missing a bunch of .woff2 files: http://screencast.com/t/kGW9xHwCL

Different PR, yeah. Let's take the discussion to IRC. cc @darkwing 

> Our production assets docs instruction to restart apache could be explicit: sudo apache2ctl restart ?

Updated

> I noticed I have/had a static/assets/ directory in my box, but it seems unnecessary? I deleted it completely and all tests still run fine in both modes. Is there some static/ cleanup to do after this change?

Hm. I can't seem to reproduce that. Can you think of another way it might have ended up there? I don't want to write it off, but it also shouldn't cause any problems if it is there, so maybe we can just keep an eye on it.
